widgetssdk/src/main/java/com/glia/widgets/callvisualizer/controller/VisitorCodeController.kt

```diff
@@ -40,14 +44,24 @@ class VisitorCodeController(
             .subscribe(
                 { visitorCode ->
                     view.showVisitorCode(visitorCode)
-                    view.setTimer(visitorCode.duration)
+                    view.setTimer(failGuardDuration(visitorCode))
                 },
                 { error ->
                     view.showError(error)
                 }
             )
     }
 
+    private fun failGuardDuration(visitorCode: VisitorCode) : Long {
+        return visitorCode.duration.run {
+            if (this <= 0) {
+                defaultRefreshDuration
+            } else {
+                this
+            }
+        }
```

Comment:
I really like the approach suggested by @yurii-glia .
what do you think about throwing an `exception` if the duration is negative

Reply:
This is a very siriouse question )
@mikktammeglia

Reply:
in some ways I agree, but this is for now just to get reoccurring web accesses to a minimum. Lets discuss this separately. I will make a task to the backlog

Reply:
I don't see the benefits of throwing an exception because:
- in this case, an integrator would need to handle this exception
- it means extra work for an integrator, additional risk for missing the handling
- anyway, integration will need to show some error message, however, why cannot we show it on our own, or just handle it silently as this PR supposes?

Maybe I'm missing anything, but as of now, I would not throw an esception.